src/node/audio_buffer_source.rs

```diff
@@ -333,7 +384,9 @@ struct AudioBufferSourceRenderer {
     stop_time: f64,
     offset: f64,
     duration: f64,
-    controller: Controller,
+    loop_: bool,
```

Comment:
I like the idea of keeping the atomics out of the render thread as much as possible.

Ideally, the render thread should only store public output values in atomics for monitoring live values from external threads. With only a single writer there won't be any race conditions and all changes are repeatable.

The render thread should not load inputs (= control values) from atomics that could change unexpectedly when being modified by multiple writers concurrently. Modifications of input values should usually performed by submitting messages. This ensures that the derived internal state remains consistent. Hidden internal state does not need to be stored in atomics, only the public output values as outlined above.

Reply:
agreed! We should verify this for all nodes that need to be changed still #315